app/commands/v2/discard_draft.rb

```diff
@@ -6,6 +6,7 @@ def call
 
         check_version_and_raise_if_conflicting(document, payload[:previous_version])
 
+        save_document_type
```

Comment:
This is because the draft is deleted before the information is required right? It might be useful to add a comment in here to mention this if this is indeed the reason.

Reply:
Good idea. That is the reason - I've amended this with a comment above the method.